src/charm.py

```diff
@@ -183,11 +183,21 @@ def healthy(self) -> bool:
 
         return True
 
-    def _on_update_status(self, _: EventBase) -> None:
+    def _on_update_status(self, event: EventBase) -> None:
         """Handler for `update-status` events."""
         if not self.healthy:
             return
 
+        # NOTE: integration with kafka-broker-rack-awareness charm.
+        # Load current properties set in the charm workload and check
+        # if rack.properties file exists
+        properties = safe_get_file(self.kafka_config.server_properties_filepath)
+        if properties:
+            if self.kafka_config.rack_properties != [] and (
```

Comment:
**thought** should we put the first if statement in the second condition, like:

` if self.kafka_config.rack_properties != [] and set(self.kafka_config.server_properties) ^ (set(properties or [])):

(or something along this line, I have not tested). this way we would run config_change even if the file does not exists. I don't know if this is a use-case could ever happen (i.e. update_status runs but the file does not exits). Maybe I'm overly conservative, but if there are no drawback, I would not see why not.

Reply:
This is done like this specifically to avoid calling `config_changed` more times than necessary. The method restarts the broker, and we would not want that happening randomly each time `update_status` trigger.